### DISCUSSION TITLE
SET button auto release

### DISCUSSION THREAD

**Discussion by eturpault** [GrandOrgue/grandorgue#1684] [General]
Once the SET button is pressed, it stays pressed until a new press action done on it. Due to that, one can by error overwrite combinations by pressing on generals / divisional / sequencer buttons (to recall a combination) whereas the SET button is still pressed and the user forgot to release it (what happened to me recently).
What do you think that the SET button is automatically released by GrandOrgue once a setter button has been pressed to memorize a combination in it ?

**Comment by ClarkeBrunt** [comment]
Personally, I think that the 'SET' action is normally programmed onto a button on a physical console. You normally hold the button down to activate set mode, and release the button to exit set mode. Of course you could program the physical button to function in other ways, like press to enter set mode, and press again to exit (like you might program, say, a Great to Pedal button). You can't so easily forget that you are holding a button down on the console. I imagine things would get confusing if SET mode was exited automatically while you were still holding the button down, indeed it might toggle the mode the 'wrong way round' (i.e. setting the mode when the button was released).

**Reply by eturpault** [reply]
Yes this is the behavior in case of a physical set button, here I am speaking about the software set button in the panels of GO, actionned by the mouse button click or a touch push. This button has no spring to release it by itself when it is no more pressed as it is the case with a physical button, so I am proposing that GO manages automatically the button releasing once a combination recording is completed (on push on a general/divisional/sequencer setter button), it would give a behavior closer to the one of a physical button (that the user releases once the combination recording is done), compared to what we have today, and would avoid to let the button pressed by forgetting and overwrite other combinations.

**Comment by larspalo** [comment]
@eturpault I personally find it quite useful that the SET is active until manually de-activated on the panel. I usually do the setter programming so that I start programming with SET active and for the first combination push CURRENT (or the equivalent button in the toolbar). Then I adjust the active stops for the next combination and push NEXT and continue programming just like that until the whole piece is done before I finally de-activate the SET. This of course works best when I know exactly what I want to have programmed... I would personally _definitely not_ want the SET to deactivate on the panel/toolbar after the Current or Next was pushed as it would increase the number of mouse clicks!

If one wants to mimic a real organ behaviour with MIDI signals from a momentary push button on the organ console it's very easy to do. I for instance have a button sending 9X Note in my setup for the SET button which makes things behave exactly like on a real organ where I have to manually hold down the SET and push another button of some kind to store the combination.

Something that I feel many people are overlooking nowadays is that GO was always considered to be played from some kind of MIDI keyboard/console setup, it was never thought of to be "complete" when running on it's own on the computer.

**Comment by eturpault** [comment]
@larspalo Indeed in the use case that you describe the SET button has to be stay pressed until the user decides to release it. There are various uses cases. 
Let's keep the current implementation unchanged, this discussion thread was only to submit a proposal and have an exchange about it.

**Comment by ahall41** [comment]
> Something that I feel many people are overlooking nowadays is that GO was always considered to be played from some kind of MIDI keyboard/console setup, it was never thought of to be "complete" when running on it's own on the computer.

OK if your console has a SET button ... that works (mine has one, but it doesn't transmit a midi signal). And I think this may be a useful _option_. I haven't used pistons very much, but usually forget to disable SET and end up overwriting the combination before I've even used it ...
